zokrates_core/src/embed.rs

```diff
@@ -66,12 +86,135 @@ impl FlatEmbed {
     /// Actually get the `FlatFunction` that this `FlatEmbed` represents
     pub fn synthetize<T: Field>(&self) -> FlatFunction<T> {
         match self {
+            #[cfg(feature = "bellman")]
+            FlatEmbed::Sha256Round => sha256_round(),
             FlatEmbed::Unpack(bitwidth) => unpack_to_bitwidth(*bitwidth),
             _ => unreachable!(),
         }
     }
 }
 
+// util to convert a vector of `(variable_id, coefficient)` to a flat_expression
+// we build a binary tree of additions by splitting the vector recursively
+#[cfg(feature = "bellman")]
+fn flat_expression_from_vec<T: Field, E: Engine>(v: &[(usize, E::Fr)]) -> FlatExpression<T> {
+    match v.len() {
+        0 => FlatExpression::Number(T::zero()),
+        1 => {
+            let (key, val) = v[0].clone();
+            let mut res: Vec<u8> = vec![];
+            val.into_repr().write_le(&mut res).unwrap();
+            FlatExpression::Mult(
+                box FlatExpression::Number(T::from_byte_vector(res)),
+                box FlatExpression::Identifier(FlatVariable::new(key)),
+            )
+        }
+        n => {
+            let (u, v) = v.split_at(n / 2);
+            FlatExpression::Add(
+                box flat_expression_from_vec::<T, E>(u),
+                box flat_expression_from_vec::<T, E>(v),
+            )
+        }
+    }
+}
+
+#[cfg(feature = "bellman")]
+impl<T: Field, E: Engine> From<BellmanConstraint<E>> for FlatStatement<T> {
+    fn from(c: BellmanConstraint<E>) -> FlatStatement<T> {
+        let rhs_a = flat_expression_from_vec::<T, E>(&c.a);
+        let rhs_b = flat_expression_from_vec::<T, E>(&c.b);
+        let lhs = flat_expression_from_vec::<T, E>(&c.c);
+
+        FlatStatement::Condition(lhs, FlatExpression::Mult(box rhs_a, box rhs_b))
+    }
+}
+
+/// Returns a flat function which computes a sha256 round
+///
+/// # Remarks
+///
+/// The variables inside the function are set in this order:
+/// - constraint system variables
+/// - arguments
+#[cfg(feature = "bellman")]
+pub fn sha256_round<T: Field>() -> FlatFunction<T> {
+    // Define iterators for all indices at hand
+    let (r1cs, input_indices, current_hash_indices, output_indices) =
```

Comment:
This function is generic over T but calls `generate_sha256_round_constraints` with Bn256. Therefore I think we should `assert_eq!(T::id(), Bn128Field::id())` here

Reply:
nice catch